test/functional/api/vim_spec.lua

```diff
@@ -1457,6 +1457,17 @@ describe('API', function()
       command("enew") -- edit new buffer, window option is reset
       eq('', nvim('get_option_value', 'colorcolumn', {scope = 'local'}))
     end)
+
+    it('get buffer or window-local options', function()
+      nvim('command', 'new')
+      local buf = nvim('get_current_buf').id
+      nvim('buf_set_option', buf, 'tagfunc', 'foobar')
+      eq('foobar', nvim('get_option_value', 'tagfunc', {buf = buf}))
+
+      local win = nvim('get_current_win').id
+      nvim('win_set_option', win, 'number', true)
+      eq(true, nvim('get_option_value', 'number', {win = win}))
```

Comment:
I think it is worth adding some coverage for equiv of `setlocal <winopt>` (`{win=win, scope='local'}`) as that is the main difference to `nvim_win_get_option` and one of the main motivations for adding this change.